src/app/Media/IBaseMediaSession.cs

```diff
@@ -0,0 +1,86 @@
+﻿//-----------------------------------------------------------------------------
+// Filename: SDPMessageMediaFormat.cs
+//
+// Description: Contains enums and helper classes for common definitions
+// and attributes used in SDP payloads.
+//
+// Author(s):
+// Jacek Dzija
+// Mateusz Greczek
+//
+// History:
+// 30 Mar 2021 Jacek Dzija,Mateusz Greczek Added MSRP
+//
+// License: 
+// BSD 3-Clause "New" or "Revised" License, see included LICENSE.md file.
+//-----------------------------------------------------------------------------
+using System.Net;
+using System.Threading.Tasks;
+using SIPSorcery.Net;
+using SIPSorcery.SIP.App;
+
+namespace SIPSorcery.app.Media
+{
+    public interface IBaseMediaSession
+    {
+        /// <summary>
```

Comment:
I understand the rationale for the splitting this interface out of IMediaSession. I think a better name for it would be ISdpSession to reflect its purpose of setting up a signalling only session with SDP.

Reply:
I don't see the need for adding another base interface under IMediaSession in this case or even changing the interface.

It seems like MSRP works together with SDP and is used as part of the SDP message and most modification happens in the SDP stack. Is there a real reason to have another interface or any need for the MSRPSession class? Can MSRP use the built-in SDP Session?


And if we can remove some methods from IMediaSession while still supporting all use-cases of the Sip Stack then maybe we should think about removing those methods from IMediaSession and adding the extended functionality in a subclass.

Reply:
@Yitzchok that was the same concern I had originally. After a closer inspection the reason to "split" the the `IMediaSession` is for a case such as `MSRP` where the "session" is text/chat only does not involve audio or video. As far as `IMediaSession` goes there should be no impact since the new `ISdpSession` or `IChatSession` is hidden under the hood.

There is no existing built in `SDP Session`.

Reply:
The IBaseMediaSession and MSRPSession implementation in this pull request can't currently be used because SIPUserAgent needs an IMediaSession regardless.

---

The point of _IMediaSession_ is to define a contract for _SIPUserAgent_ so that it can exchange SDP messages without caring about who generates the SDP message or what it contains.

It shouldn't technically matter what the **media** is (Audio/Video/**Data**).

Currently, it does have some Audio/Video specific properties as well as the SendDtmf method.

If we want to do this refactor, we have to think about SIPUserAgent which is the technical owner of the IMediaSession contract. So to support the IBaseMediaSession concept we will have to get it to work with SIPUserAgent.

Another option is to change IMediaSession to only have the core properties (like IBaseMediaSession) and add IAudioVideoMediaSession for media as a placeholder for the current additional options.

We also have to ensure that SIPUserAgent can work with the reduced IMediaSession as well as support all the existing audio/video features.

---

Regarding using ISdpSession instead of IMediaSession
SDP is a protocol describing multimedia sessions so calling it a MediaSession seems correct to me. Because it is not an SDP session rather it's a Media Session that uses SDP to describe it.

Reply:
@Yitzchok the `IMediaSession : IBaseMediaSession` in this PR does work. I built it and checked all the unit tests passed. But that aside I suspect some of your discomfort comes from the interfaces getting too fragmented or granular. At least that's what I felt.

I don't feel that strongly either way but perhaps it would be less disruptive if `IMediaSession` was left as is and `MSRPSession` inherited from it and used a `NotImplementedException` on the audio and video calls it didn't support. @jacekdzija what do you think? It's actually unlikely that the `IBaseMediaSession` being added would be used by any other classes in the near future so perhaps it's better to deal with it in `MSRPSession` and leave `IMediaSession` alone.

Reply:
I have no issue with IMediaSession becoming more granular (I think if it can be done then that would be amazing). It does have more than it should, if I remember correctly it's mainly to support the current state of SIPUserAgent.

Regarding the tests running doesn't mean it works as expected or is useable since no code here is actually using _MSRPSession_.

The problem here is that _SIPUserAgent_ still expects _IMediaSession_ so you can't pass it _MSRPSession_ which implements _IBaseMediaSession_ which doesn't derive from _IMediaSession_

Reply:
@sipsorcery @Yitzchok thanks for good comments. I agree with you and my intention was to make a more generic interface 

@Yitzchok  I have a solution that uses sipsorcery together with https://github.com/MSRP-OSS/msrp-csharp and I have voice and text calls scenarios implemented and everything works fine with real system.
SIPUserAgent has been changed and it accepts IBaseMediaSession - I missed that file during committing the change

@sipsorcery  you are right that for now only MSRP is using the base interface so I can revert this change and leave IMediaSession as it was and implement empty methods in MSRP class. 


Let's take a look once again on new commit and if you feel that IMediaSession shall be reverted I will update this PR accordingly . Just vote/or leave a comment - for me both options are okey.

Reply:
+1 for the original `IMediaSession`.

Reply:
restored - take a look thx

Reply:
@jacekdzija your latest commit 0bf3e85bc450c1eae879a8986346e8585a90fedc has put back the original `IMediaSession`. But now the `MSRPSession` class has been removed. Was that intentional? Isn't that class the main purpose for the Pull Request?

Reply:
From what I understand MSRPSession was just a stub.
The main code is the SDP changes to support MSRP.

Reply:
@Yitzchok  yes

Reply:
@sipsorcery MSRPSession was a stub to show my intention about IBaseMediaSession interface now you can implement your own using IMediaSession